src/frontend/src/flows/pin/usePin.ts

```diff
@@ -57,14 +57,19 @@ const usePinTemplate = ({
 };
 
 export const usePinPage = renderPage(usePinTemplate);
-export const usePin = (): Promise<
-  { kind: "pin"; pin: string } | { kind: "canceled" } | { kind: "passkey" }
+export const usePin = <T>({
+  verifyPin,
+}: {
+  verifyPin: (pin: string) => Promise<PinResult<T>>;
+}): Promise<
+  { kind: "pin"; result: T } | { kind: "canceled" } | { kind: "passkey" }
```

Comment:
it used to returns `{ kind: "pin"; pin: string }` but now it returns ` { kind: "pin"; result: T }`. Changing the variable name from `pin` to `result` is expected?

Reply:
Yes, because `verify` can transform the value now. I.e. the result indicates that the `pin` was successfully used to get some `result`.

I'm open to rename stuff, if it is confusing now.